kyaml/yaml/rnode.go

```diff
@@ -1063,6 +1688,7 @@ func checkKey(key string, elems []*Node) bool {
 	return count == len(elems)
 }
 
+// Deprecated: Use Get instead.
 // GetSlice returns the contents of the slice field at the given path.
 func (rn *RNode) GetSlice(path string) ([]interface{}, error) {
```

Comment:
I'm wondering why you chose to provide `GetNestedString` and `GetNestedInt` etc but no `GetNestedSlice`? I'm not saying we need to have it in this PR but is that something we would want to add?

Reply:
Answer is similar to https://github.com/kubernetes-sigs/kustomize/pull/4462#discussion_r804269357.
Returning `[]interface{}` and let the user to do the type assertion is not a great UX.
If users want something like `[]string` or `[]corev1.Container`, they can pass it in `Get()` and they get a strong typed object.